### PR TITLE
Update Identity server dependency for preview4

### PR DESCRIPTION
**Pull request by javiercn**
* Updates the identity server version to use the bits targeting 3.0.
* Updates the EF migrations.
* Updates react node package versions to latest as previous version have issues.
* Adds support for creating more than one browser instance through the browser fixture.
  * This is needed so that auth tests run in isolation.
* Adds an E2E test to cover auth.

### CODE REVIEW

File: src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/ClientApp/src/api-authorization/authorize.service.ts [hunk 1]
```diff
@@ -50,6 +50,10 @@ interface IAuthenticationState {
   providedIn: 'root'
 })
 export class AuthorizeService {
+  // By default pop ups are disabled because they don't work properly on Edge.
+  // If you want to enable pop up authentication simply remove this line of code
```
Comment (pranavkm) [inline src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/ClientApp/src/api-authorization/authorize.service.ts]: > simply set this value to `false`

Doesn't really matter, but it's less work for the user

File: src/ProjectTemplates/Web.Spa.ProjectTemplates/content/React-CSharp/ClientApp/src/components/api-authorization/AuthorizeService.js [hunk 2]
```diff
@@ -7,6 +7,12 @@ export class AuthorizeService {
     _user = null;
     _isAuthenticated = false;
 
+    // By default pop ups are disabled because they don't work properly on Edge.
+    // If you want to enable pop up authentication simply remove this line of code
+    // and the associated if clauses below.
+    _popUpDisabled = true;
+
+
```
Comment (pranavkm) [inline src/ProjectTemplates/Web.Spa.ProjectTemplates/content/React-CSharp/ClientApp/src/components/api-authorization/AuthorizeService.js]: Extra line

**Review by pranavkm — APPROVED** [review]

File: eng/Versions.props [hunk 3]
```diff
@@ -177,9 +177,11 @@
...
     <GrpcCorePackageVersion>1.19.0-pre1</GrpcCorePackageVersion>
     <GrpcToolsPackageVersion>1.19.0-pre1</GrpcToolsPackageVersion>
-    <IdentityServer4AspNetIdentityPackageVersion>2.3.0</IdentityServer4AspNetIdentityPackageVersion>
-    <IdentityServer4EntityFrameworkPackageVersion>2.3.0</IdentityServer4EntityFrameworkPackageVersion>
-    <IdentityServer4PackageVersion>2.3.0</IdentityServer4PackageVersion>
+    <IdentityServer4AspNetIdentityPackageVersion>3.0.0-preview3.4</IdentityServer4AspNetIdentityPackageVersion>
+    <IdentityServer4EntityFrameworkPackageVersion>3.0.0-preview3.4</IdentityServer4EntityFrameworkPackageVersion>
+    <IdentityServer4PackageVersion>3.0.0-preview3.4</IdentityServer4PackageVersion>
+    <IdentityServer4StoragePackageVersion>3.0.0-preview3.4</IdentityServer4StoragePackageVersion>
+    <IdentityServer4EntityFrameworkStoragePackageVersion>3.0.0-preview3.4</IdentityServer4EntityFrameworkStoragePackageVersion>
```
Comment (dougbu) [inline eng/Versions.props]: @Eilon please confirm these new / updated dependencies are fine.

**Review by dougbu — COMMENTED** [review]
eng/ changes look fine to me but please check w/ @Eilon

File: eng/Versions.props (hunk 3 above)
Comment (javiercn) [reply eng/Versions.props]: They are.

File: src/Identity/ApiAuthorization.IdentityServer/samples/ApiAuthSample/Pages/Index.cshtml [hunk 4]
```diff
@@ -9,6 +9,7 @@
 <!DOCTYPE html>
 <html>
 <head>
+    <base href="~/" />
```
Comment (Eilon) [inline src/Identity/ApiAuthorization.IdentityServer/samples/ApiAuthSample/Pages/Index.cshtml]: Just checking: Does the tag helper stuff properly resolve a `<base href="...">` element like other link elements? I've never tried that so I'm not sure.

Comment (javiercn) [reply src/Identity/ApiAuthorization.IdentityServer/samples/ApiAuthSample/Pages/Index.cshtml]: Yeah, our blazor templates do it this way too.

**Review by ryanbrandenburg — APPROVED** [review]
Seems fine.

**Pull request merged**